include/Group.hh

```diff
@@ -20,10 +22,8 @@ public:
 
   // Group operations
   virtual E mul(const E& a, const E& b) = 0;
-  virtual E exp(const E& a, mpz_t n) = 0;
-
-  // I/O methods
-  virtual std::string toString() const = 0;
+  virtual E exp(const E& a, const E& n) = 0;
+  virtual BigInt order() = 0;
```

Comment:
A group can have an arbitrarily large order. So, for the return type of this method, we need something that stores arbitrarily large numbers.